ssh_check/tests/test_ssh.py

```diff
@@ -177,15 +192,24 @@ def test_force_sha1_disabled(aggregator, dd_run_check, settings):
     ]
 
 
-def test_force_sha1_enabled(aggregator, dd_run_check):
+@pytest.mark.parametrize(
+    'authenticated,service_check_result',
+    [
+        pytest.param(True, CheckSSH.OK, id='authenticated'),
+        pytest.param(False, CheckSSH.CRITICAL, id='not authenticated'),
+    ],
+)
+def test_force_sha1_enabled(aggregator, dd_run_check, authenticated, service_check_result):
     settings = {'force_sha1': True}
     inst = deepcopy(common.INSTANCES['main'])
     inst.update(settings)
-    client, ssh = _setup_check_with_mock_client(inst, connect_result=CONNECTION_SUCCEEDED)
+    client, ssh = _setup_check_with_mock_client(
+        inst, connect_result=CONNECTION_SUCCEEDED, authentication_result=authenticated
+    )
```

Comment:
In my opinion, this test about forcing sha1 doesn't depend on authentication, we could just set the authentication_result to be always True. The auth behavior is already covered by `test_ssh`, the other assertions are not affected at all by that.